### PR TITLE
chore(protractor): support Protractor 5.x.x with the new webdriver-manager

### PR DESCRIPTION
**Pull request by cnishina**
- During the static call (for 5.4.2 to built/lib/config) to
    getSeleniumDir, construct the update-config.json. This is just combining
    existing information from geckodriver.config.json, chrome.config.json
    and selenium-server.config.json. Important note that since this is just
    added in, this file will not be removed during the clean up and will
    always be overwritten when making the static call. If it is called 3
    times in the driverProviders/local test, it will rewrite the file 3
    times.
- Add test for Protractor. For both version 5.4.2 and version 6+ calls
    to webdriver-manager.
- Add jasmine protractor test to circle ci test.

### CODE REVIEW

**Comment by cnishina** [comment]
JK. I looked at Protractor master branch.... I think I can fake in something to work. Maybe JS files in the lib/ that get copied to built/lib/config

**Comment by cnishina** [comment]
Okay this should be ready for review now. It creates the update-config.json file from the existing config files. The directories match: `built/lib/` with a `config.js` file. And the `update-config.json` file when read and parsed to json has `updateConfig.chrome.last`, `updateConfig.gecko.last`, and `updateConfig.standalone.last`.

Note this always rewrites the file on every static call. I could check to see the age of the file but the max rewrite is 3 times. This isn't the best fix and will make it better if there is a problem.

File: lib/legacy/config.spec-ptor.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { Config } from './config';
+
+describe('config', () => {
+  //let updateJson = path.resolve(SeleniumConfig.getSeleniumDir(), 'update-config.json');
```
Comment (heathkit) [inline lib/legacy/config.spec-ptor.ts]: Should this just be deleted?

File: lib/legacy/config.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { JsonObject } from '../provider/utils/http_utils';
+
+// Need to support this call in Protractor:
+// const SeleniumConfig = require('webdriver-manager/built/lib/config').Config;
+// During the publish phase, we will need to move dist/legacy/config.* to
+// to built/lib/config.
+export class Config {
+  // SeleniumConfig.getSeleniumDir()
```
Comment (heathkit) [inline lib/legacy/config.ts]: Unnecessary comment

File: lib/legacy/config.spec-ptor.ts (hunk 1 above)
Comment (cnishina) [reply lib/legacy/config.spec-ptor.ts]: Done.

File: lib/legacy/config.ts (hunk 2 above)
Comment (cnishina) [reply lib/legacy/config.ts]: Done.

**Pull request merged**